packages/app-desktop/gui/MainScreen/commands/perNotebookSortOrder.ts

```diff
@@ -0,0 +1,212 @@
+	loadOwnSortOrders();
+	loadSharedSortOrder();
+	eventManager.appStateOn('notesParentType', onNotebookSelectionMayChange);
+	eventManager.appStateOn('selectedFolderId', onNotebookSelectionMayChange);
+
+	return {
+		enabledCondition: 'oneFolderSelected',
+
+		execute: async (context: CommandContext, notebookId?: string, own?: boolean) => {
+			let targetId = notebookId;
+			const selectedId = getSelectedNotebookId(context.state);
+			if (!targetId) {
+				targetId = selectedId; // default: selected notebook
+				if (!targetId) return;
+			}
+			const targetOwn = hasOwnSortOrder(targetId);
+			let newOwn;
+			if (typeof own == 'undefined') {
+				newOwn = !targetOwn; // default: toggling
+			} else {
+				newOwn = !!own;
+				if (newOwn === targetOwn) return;
+			}
+			if (newOwn) {
+				let field: string, reverse: boolean;
+				if (!hasOwnSortOrder(selectedId)) {
+					field = getCurrentField();
+					reverse = getCurrentReverse();
+				} else {
+					field = sharedSortOrder.field;
+					if (perFieldReversalEnabled()) {
+						reverse = sharedSortOrder[field];
+					} else {
+						reverse = sharedSortOrder.reverse;
+					}
+				}
+				setOwnSortOrder(targetId, field, reverse);
+			} else {
+				deleteOwnSortOrder(targetId);
+			}
+		},
+	};
+};
+
+function onNotebookSelectionMayChange() {
+	const selectedId = getSelectedNotebookId();
+	if (previousNotebookId === null) previousNotebookId = selectedId;
+	if (previousNotebookId === selectedId) return;
+	const field = getCurrentField();
+	const reverse = getCurrentReverse();
+	const previousNotebookHasOwnSortOrder = hasOwnSortOrder(previousNotebookId);
+	if (previousNotebookHasOwnSortOrder) {
+		setOwnSortOrder(previousNotebookId, field, reverse);
+	} else {
+		setSharedSortOrder(field, reverse);
+	}
+	previousNotebookId = selectedId;
+	let next;
+	if (hasOwnSortOrder(selectedId)) {
+		next = getOwnSortOrder(selectedId);
+	} else if (previousNotebookHasOwnSortOrder) {
+		next = sharedSortOrder;
+	} else {
+		return;
+	}
+	if (perNotebookSortOrderEnabled()) {
+		if (next.field != field || next.reverse != reverse) {
+			void CommandService.instance().execute(NOTES_SORT_ORDER_SWITCH, next.field, next.reverse);
+		}
+	}
+}
+
+const OWN_SORT_ORDERS = 'notes.perNotebookSortOrders';
+const SHARED_SORT_ORDER = 'notes.sharedSortOrder';
+const SUFFIX_FIELD = '$field';
+const SUFFIX_REVERSE = '$reverse';
+
+function getSelectedNotebookId(state?: State): string {
+	const s = state ? state : app().store().getState();
+	if (s.notesParentType === 'Folder') {
+		return s.selectedFolderId;
+	} else {
+		return '';
+	}
+}
+
+function getCurrentField(): string {
+	return Setting.value(SETTING_FIELD);
+}
+
+function getCurrentReverse(): boolean {
+	return Setting.value(SETTING_REVERSE);
+}
+
+function perFieldReversalEnabled(): boolean {
+	return Setting.value(SETTING_PER_FIELD_REVERSAL_ENABLED);
+}
+
+function perNotebookSortOrderEnabled(): boolean {
+	return Setting.value(SETTING_PER_NOTEBOOK_SORT_ORDER_ENABLED);
+}
```

Comment:
I don't think we need this layer of indirection. In general, the code should just get the setting directly, without going through a function (unless of course the function does additional processing, but that's not happening in this case). Although I realise you lose typing in this case, but then we should probably change `Setting.value` to a `Setting.value<T>`

Reply:
I see.

Reply:
finished. 4cdba95